google/cloud/storage/testing/retry_tests.cc

```diff
@@ -81,6 +85,36 @@ ::testing::Matcher<Status> StoppedOnPermanentError(char const* api_name) {
       matches_api_name(api_name), matches_exhausted());
 }
 
+::testing::Matcher<std::vector<std::string>> RetryLoopUsesSingleToken() {
+  return ::testing::AllOf(
+      Not(IsEmpty()),
+      ResultOf(
+          "The retry loop uses a non-empty token",
+          [](std::vector<std::string> const& v) {
+            return v.empty() ? std::string{} : v.front();
+          },
+          Not(IsEmpty())),
+      ResultOf(
+          "All tokens are equal",
+          [](std::vector<std::string> const& v) { return v; },
+          Truly([](std::vector<std::string> const& v) {
+            auto const s = v.empty() ? std::string{} : v.front();
+            return std::all_of(v.begin(), v.end(),
+                               [s](auto const& x) { return x == s; });
+          })));
+}
+
+void CaptureIdempotencyToken(std::vector<std::string>& tokens,
+                             rest_internal::RestContext const& context) {
```

Comment:
Q.  Do we have (or inherit) any preference for inputs/outputs ordering?

Reply:
Not really.  I am happy to swap them if it makes the code more readable, or more consistent with something...

Reply:
Nah, I'm happy without such a convention.